src/domain.rs

```diff
@@ -33,7 +36,7 @@ loom::lazy_static! {
 trait WithMut<T> {
     fn with_mut<R>(&mut self, f: impl FnOnce(&mut *mut T) -> R) -> R;
 }
-impl<T> WithMut<T> for std::sync::atomic::AtomicPtr<T> {
+impl<T> WithMut<T> for crate::sync::atomic::AtomicPtr<T> {
```

Comment:
```suggestion
impl<T> WithMut<T> for core::sync::atomic::AtomicPtr<T> {
```
I think this was intentionally _not_ the one from `crate::sync`.